File: frontend/utils/validateDomain.ts
const domainRegex = /^(?:[a-zA-Z0-9][a-zA-Z0-9-]{0,62}\.)+[a-zA-Z]{2,63}$/;

export function validateDomain(domain: string): boolean {
  if (!domain) return false;

  if (domain.length < 1 || domain.length > 253) {
    return false;
  }

  if (
    domain.startsWith("-") ||
    domain.endsWith("-") ||
    domain.startsWith("http") ||
    domain.startsWith("https") ||
    domain.startsWith("www")
  ) {
    return false;
  }

  return domainRegex.test(domain);
}

File: backend/internal/helpers/validateDomain.go
package helpers

import (
	"regexp"
	"strings"
)

var domainRegex = regexp.MustCompile(`^(?:[a-zA-Z0-9][a-zA-Z0-9-]{0,62}\.)+[a-zA-Z]{2,63}$`)

func ValidateDomain(domain string) bool {
	if len(domain) < 1 || len(domain) > 253 {
		return false
	}
	if strings.HasPrefix(domain, "-") || strings.HasSuffix(domain, "-") {
		return false
	}
	return domainRegex.MatchString(domain)
}
